File: SourceCppProject/sources/Parsers/GeneralCharacterValidator.cpp
#include "GeneralCharacterValidator.hpp"

#include <cctype>


bool GeneralCharacterValidator::isNotCommentCharacter(const char& character)
{
    const char comment_character {'#'};

    if (character == comment_character)
    {
        return false;
    }

    return true;
}

bool GeneralCharacterValidator::isSpaceCharacter(const char& character)
{
    return std::isspace(character);
}

File: TextLinesComparing.Library/GeneralCharacterValidator.cs
namespace TextLinesComparing.Library;

public static class GeneralCharacterValidator
{
    public static bool IsNotCommentCharacter(char character)
    {
        const char comment_character = '#';

        if (character == comment_character)
        {
            return false;
        }

        return true;
    }

    public static bool IsSpaceCharacter(char character)
    {
        return char.IsWhiteSpace(character);
    }
}
